### ISSUE TITLE
Precision and mAP not high enough

### ISSUE THREAD

**Issue by brys17** [ultralytics/yolov5#11766]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

Hi everyone! I am new to YOLOv5 and this is my first time using it. I have trained my own custom dataset regarding tea leaves diseases. I have trained using multiple YOLOv5 (n,s,m,l) and got the result of only 50-60%. Does anyone know what is the problem and how to improve the results?

Here are the parameters of my training:
- Classes: 3 (algal leaf spot, brown blight, grey blight)
- Labelled using Roboflow
- Total training images: 850 (2000 images after augmentation)
- Testing images: 150
- Trained using Colab
- !python train.py --img 640 --batch 16 --epochs 100 --data data.yaml --weights yolov5m.pt --cache

Here are the results I got
![confusion_matrix](https://github.com/ultralytics/yolov5/assets/137619824/148f39ed-3b32-4e99-8069-cafccf6d0b0b)
![F1_curve](https://github.com/ultralytics/yolov5/assets/137619824/8128dc8e-405f-479c-a5f1-977c92d1696e)
![results](https://github.com/ultralytics/yolov5/assets/137619824/17fea9d2-6818-494c-9ab3-92edfa5a9c76)
![val_batch1_pred](https://github.com/ultralytics/yolov5/assets/137619824/485a284a-f5f1-4979-9065-7c194264b193)

And here are the example of the dataset images
![IMG_20230612_152201](https://github.com/ultralytics/yolov5/assets/137619824/c2fbc46f-e3f1-43cf-96d0-15ff9c502415)
![IMG_20230612_152924](https://github.com/ultralytics/yolov5/assets/137619824/8f9d35ca-ee25-4fc4-bf34-f199a2cb6665)
![IMG_20230612_153520](https://github.com/ultralytics/yolov5/assets/137619824/3c4b0cea-1399-4f59-a8c8-6736797dfdaf)
![IMG_20230612_152424](https://github.com/ultralytics/yolov5/assets/137619824/6fe84ba3-892e-412b-9c0d-dc35af23a20f)
![IMG_20230612_153537](https://github.com/ultralytics/yolov5/assets/137619824/ebfe873b-6652-4cf9-809e-582320a4313c)
![IMG_20230612_154643](https://github.com/ultralytics/yolov5/assets/137619824/f4f1c698-5aa7-4aa8-90b1-86c525131035)
![IMG_20230612_152145](https://github.com/ultralytics/yolov5/assets/137619824/d8b4a8ce-9150-4635-8f08-cdb387462edc)
![IMG_20230612_152935](https://github.com/ultralytics/yolov5/assets/137619824/2f6b7226-00f4-40a1-9d63-ae757abb2b6c)
![IMG_20230612_153332](https://github.com/ultralytics/yolov5/assets/137619824/545f5165-354e-4207-9fbb-12df6ee929ac)

### Additional

Any help will be appreciated! Thank you!

**Comment by glenn-jocher** [comment]
@brys17 hi there! Thank you for reaching out and providing detailed information about your training setup. 

Based on the information you provided, here are a few suggestions to improve the results of your YOLOv5 model:

1. **Increase the number of training epochs**: Training for more epochs might allow the model to learn more and improve performance. You can try increasing the number of epochs and observe if it leads to better results.

2. **Try different anchor box settings**: Experimenting with anchor box sizes can improve detection accuracy for different object sizes. You might want to adjust the anchor sizes in the `yolov5/models/yolov5m.yaml` file and retrain the model to see if it helps capture the tea leaves diseases better.

3. **Check the quality of the labeled dataset**: Make sure that the tea leaves and their diseases are properly labeled in your dataset. Inspecting the annotations and ensuring they align with the tea leaves' actual locations can significantly impact the performance of the model.

4. **Explore other training techniques**: You can try different techniques like mixup, mosaic augmentation, or using focal loss to improve the model's ability to handle overlapping or small objects. The YOLOv5 repository provides options to enable these techniques during training.

5. **Consider model architecture**: YOLOv5 comes in different sizes (s, m, l, and x), and choosing the right architecture can impact performance. You can experiment with different architectures and assess their impact on your task.

It's worth noting that improving accuracy might require some fine-tuning and iterations. You may need to try different combinations of the above suggestions, experiment with hyperparameters, and potentially gather more training data if necessary.

Lastly, I recommend checking out the [YOLOv5 GitHub Discussions](https://github.com/ultralytics/yolov5/discussions) and [GitHub Issues](https://github.com/ultralytics/yolov5/issues) as they contain valuable insights from the YOLO community. Others may have faced similar challenges and found effective solutions that could benefit you.

I wish you the best of luck with your tea leaves disease detection project! Let us know if you have any further questions or need additional assistance.

**Comment by brys17** [comment]
Hi there @glenn-jocher ! Thank you for your suggestions! I will try to apply it on my work

**Comment by glenn-jocher** [comment]
@brys17 you're very welcome! If you have any more questions or need further assistance as you apply these suggestions, feel free to ask. Best of luck with your work, and I hope the improvements yield the results you're looking for!
